File: Sources/RMBTLoopModeConfirmationViewController.m
/*
 * Copyright 2017 appscape gmbh
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */

#import "RMBTLoopModeConfirmationViewController.h"
#import "WKWebView+RMBTConfiguration.h"

@interface RMBTLoopModeConfirmationViewController () {
    BOOL _step2;
}

@property (weak, nonatomic) IBOutlet UIButton *declineButton;
@property (weak, nonatomic) IBOutlet UIButton *acceptButton;

@end

@implementation RMBTLoopModeConfirmationViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self.acceptButton setTitle:NSLocalizedString(@"text_button_accept", @"") forState:UIControlStateNormal];
    [self.declineButton setTitle:NSLocalizedString(@"text_button_decline", @"") forState:UIControlStateNormal];
    self.acceptButton.layer.cornerRadius = 8;
    self.declineButton.layer.cornerRadius = 8;
    [self createWebView];
    [self show];
}

- (void)createWebView {
    WKWebView *webView = [WKWebView wideWebViewWithFrame:self.view.bounds];
    webView.translatesAutoresizingMaskIntoConstraints = NO;
    [self.view addSubview:webView];
    self.webView = webView;
    
    [NSLayoutConstraint activateConstraints:@[
        [webView.leftAnchor constraintEqualToAnchor:self.view.leftAnchor],
        [webView.rightAnchor constraintEqualToAnchor:self.view.rightAnchor],
        [webView.topAnchor constraintEqualToAnchor:self.view.topAnchor],
        [webView.bottomAnchor constraintEqualToAnchor:self.toolbar.topAnchor]
    ]];
}

- (void)show {
    self.navigationController.navigationBar.barStyle = UIBarStyleBlack;
    
    self.navigationItem.prompt = nil;
    self.navigationItem.title = NSLocalizedString(@"title_loop_instruction_1", @"Confirmation dialog title 1/2");
    NSString *html = @"loop_mode_info";

    if (_step2) {
        self.navigationItem.title = NSLocalizedString(@"title_loop_instruction_2", @"Confirmation dialog title 1/2");
        html = @"loop_mode_info2";
    }

    NSURL* url = [NSURL fileURLWithPath:[[NSBundle mainBundle] pathForResource:html ofType:@"html"]];
    [self.webView loadRequest:[NSURLRequest requestWithURL:url]];
}

- (IBAction)accept:(id)sender {
    if (!_step2) {
        _step2 = YES;
        [self show];
    } else {
        [self performSegueWithIdentifier:@"accept" sender:self];
    }
}

@end

File: Sources/RMBTLoopModeConfirmationViewController.swift
//
//  RMBTLoopModeConfirmationViewController.swift
//  RMBT
//
//  Created by Sergey Glushchenko on 08.12.2021.
//  Copyright © 2021 appscape gmbh. All rights reserved.
//

import UIKit

final class RMBTLoopModeConfirmationViewController: UIViewController {
    private let acceptSegue = "accept"
    
    @IBOutlet weak var toolbar: UIView!
    @IBOutlet weak var declineButton: UIButton!
    @IBOutlet weak var acceptButton: UIButton!
    
    lazy var webView: WKWebView = {
        let webView = WKWebView.wideWebView(with: self.view.bounds)
        webView.translatesAutoresizingMaskIntoConstraints = false
        return webView
    }()
    
    private var isStep2 = false
    
    override func viewDidLoad() {
        super.viewDidLoad()

        self.acceptButton.setTitle(.accept, for: .normal)
        self.declineButton.setTitle(.decline, for: .normal)
        
        self.acceptButton.layer.cornerRadius = 8
        self.declineButton.layer.cornerRadius = 8
        self.createWebView()
        self.show()
    }
    
    private func createWebView() {
        self.view.addSubview(webView)
        
        NSLayoutConstraint.activate([
            webView.leftAnchor.constraint(equalTo: self.view.leftAnchor),
            webView.rightAnchor.constraint(equalTo: self.view.rightAnchor),
            webView.topAnchor.constraint(equalTo: self.view.topAnchor),
            webView.bottomAnchor.constraint(equalTo: self.toolbar.topAnchor)
        ])
        
        self.view.bringSubviewToFront(toolbar)
    }


    private func show() {
        self.navigationController?.navigationBar.barStyle = .black
        
        self.navigationItem.prompt = nil
        self.navigationItem.title = .title1
        var html = "loop_mode_info"

        if isStep2 {
            self.navigationItem.title = .title2
            html = "loop_mode_info2"
        }

        guard let path = Bundle.main.path(forResource: html, ofType: "html") else { return }
        let url = URL(fileURLWithPath: path)
        self.webView.load(URLRequest(url: url))
    }
    
    @IBAction func accept(_ sender: Any) {
        if !isStep2 {
            isStep2 = true
            self.show()
        } else {
            self.performSegue(withIdentifier: acceptSegue, sender: self)
        }
    }
}

private extension String {
    static let accept = NSLocalizedString("text_button_accept", comment: "")
    static let decline = NSLocalizedString("text_button_decline", comment: "")
    
    static let title1 = NSLocalizedString("title_loop_instruction_1", comment: "Confirmation dialog title 1/2")
    static let title2 = NSLocalizedString("title_loop_instruction_2", comment: "Confirmation dialog title 2/2")
}
